pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -341,7 +372,57 @@ func determineCandidatesWithSameRegionStrategy(seedList []gardencorev1beta1.Seed
 	return candidates
 }
 
-func determineCandidatesWithMinimalDistanceStrategy(seeds []gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) []gardencorev1beta1.Seed {
+func determineCandidatesWithMinimalDistanceStrategy(log logr.Logger, shoot *gardencorev1beta1.Shoot, seedList []gardencorev1beta1.Seed, schedulerRegionConfig *corev1.ConfigMap) ([]gardencorev1beta1.Seed, error) {
+	candidates, err := regionConfigMinimalDistance(log, seedList, shoot, schedulerRegionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fall back to Levenshtein minimal distance in case we didn't find any candidates.
+	if len(candidates) == 0 {
+		log.Info("No candidates found with minimal distance of region config. Falling back to Levenshtein minimal distance")
+		candidates = levenshteinMinimalDistance(seedList, shoot)
+	}
+	return candidates, nil
+}
+
+func regionConfigMinimalDistance(log logr.Logger, seeds []gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot, schedulerRegionConfig *corev1.ConfigMap) ([]gardencorev1beta1.Seed, error) {
+	var candidates []gardencorev1beta1.Seed
+
+	if schedulerRegionConfig == nil || schedulerRegionConfig.Data[shoot.Spec.Region] == "" {
+		log.Info("Shoot region not available in configmap", "shootRegion", shoot.Spec.Region)
+		return candidates, nil
+	}
+
+	regionConfig := make(map[string]int)
+	if err := yaml.Unmarshal([]byte(schedulerRegionConfig.Data[shoot.Spec.Region]), &regionConfig); err != nil {
+		return nil, fmt.Errorf("failed to determine seed candidates. Wrong format in ConfigMap %s/%s, Region %s: %w", schedulerRegionConfig.Namespace, schedulerRegionConfig.Name, shoot.Spec.Region, err)
+	}
+
+	minDistance := math.MaxInt32
+	for _, seed := range seeds {
+		dist, ok := regionConfig[seed.Spec.Provider.Region]
+		if !ok {
+			log.Info("Seed region not found in config", "seedRegion", seed.Spec.Provider.Region)
```

Comment:
Would it make sense to add the name of the seed as well. Otherwise, we might simply see some regions multiple times here.

Reply:
We can but it probably won't help much to fix the issue since one only needs the region?

Reply:
Sure, the region is enough to identify the missing region. However, now we will have potentially one message multiple times depending on the number of seeds in the region.
I would prefer to have the message only once or at least have somewhat different message, e.g. by adding the seed name.